### PR TITLE
Bugfix 22987 - Game filter does not work.

### PR DESCRIPTION
**Pull request by Randypk**
Changed the search filter in the mulitplayer lobby to
search on era and mod as well as title and description.

### CODE REVIEW

File: src/game_initialization/multiplayer_lobby.cpp [hunk 1]
```diff
@@ -483,7 +483,7 @@ void gamebrowser::populate_game_item_campaign_or_scenario_info(gamebrowser::game
 
 						if (i.first == game["mp_scenario"] &&
 							i.second == hash) {
-
+
```
Comment (Vultraz) [inline src/game_initialization/multiplayer_lobby.cpp]: You added unnecessary whitespace here.

**Comment by Randypk** [comment]
Must be from a debug statement I added then removed.  I'll close this request and submit a new one.

**Comment by gfgtdf** [comment]
whitespace fixups should be rebased into the commit that introduced them if possible.

**Comment by Randypk** [comment]
Both these commits are in the same branch, I can't seem to rebase the fixup back into the first commit.

**Comment by gfgtdf** [comment]
ok merged in https://github.com/wesnoth/wesnoth/commit/dbcd4e3349101d9f167f4e0647d44e11c4595ad1
